mesonbuild/compilers/compilers.py

```diff
@@ -469,7 +469,7 @@ def sanitizer_compile_args(value):
 def sanitizer_link_args(value):
     if value == 'none':
         return []
-    args = ['-fsanitize=' + value]
+    args = ['-fsanitize=' + value, '-ldl', '-lrt', '-lm', '-lpthread']
```

Comment:
this is going to break every OS/compiler combination where -fsanitize exists but any of libdl, librt, libm, or libpthread doesn't. What is the exact problem you're trying to solve?

Reply:
```
[  501s] FAILED: c/prog 
[  501s] cc  -o c/prog 'c/2e7d2c0@@prog@exe/prog.c.o' -fsanitize=address -Wl,--no-undefined -Wl,--as-needed  
[  501s] /usr/lib/gcc/i586-suse-linux/9/../../../../i586-suse-linux/bin/ld: /usr/lib/gcc/i586-suse-linux/9/libasan.a(sanitizer_linux_libcdep.o): in function `__sanitizer::GetThreadStackTopAndBottom(bool, unsigned long*, unsigned long*)':
[  501s] /home/abuild/rpmbuild/BUILD/gcc-9.0.1+r269212/obj-i586-suse-linux/i586-suse-linux/libsanitizer/sanitizer_common/../../../../libsanitizer/sanitizer_common/sanitizer_linux_libcdep.cc:136: undefined reference to `pthread_getattr_np'
```

More details in https://bugzilla.opensuse.org/show_bug.cgi?id=1127953

Reply:
Which test is this?

Reply:
This seems more like a toolchain bug to me. If you pass `-fsanitize=address` then the compiler should do everything needed to make it work or print out a specific error message saying that it can not perform the requested operation rather than failing with generic error message.